R/ltv.R

```diff
@@ -369,5 +368,7 @@ vcov.ltv <- function(object, ...) {
 }
 
 tsdiag.ltv <- function(object, gof.lag = 10, ...) {
-  stats:::tsdiag.Arima(object, gof.lag, ...)
+  requireNamespace(stats, quietly = TRUE)
+  getS3method("tsdiag", "Arima")
+  tsdiag.Arima(object, gof.lag, ...)
 }
```

Comment:
There's several things I'm unsure about here that your testing might have addressed.
- I would have thought that `requireNamespace` is not needed for stats given that it is always loaded, at least on the search path
- As far as I understand, getS3method is a function, not a procedure, so it actually returns the function tsdiag.Arima, which then needs to be called, i.e. it would have to be `tsdiag.Arima <- getS3method("tsdiag","Arima")`

I'm happy to be corrected. This is a little tricky, hence why I've never bothered to resolve it.
I assume we don't have any tests for this function given it produces a plot. Do we have an example of it that could be used for testing?

Reply:
It might be worth adding a smoke test for this kind of thing.

Reply:
@josephguillaume you're correct because running `getS3method("tsdiag", "Arima")` returns the function, so it seems I do need to assign it. And yup, there are no tests for it at present, can set up a smoke test.

Reply:
I've updated this PR with the changes proposed above. Since `arima` only accepts a univariate time series as input, I've implemented the smoke test correctly by selecting a particular column from a dataset. @josephguillaume is this how you would use `tsdiag` with `arima`?

> I assume we don't have any tests for this function given it produces a plot. Do we have an example of it that could be used for testing?

The smoke test I've included makes sure that the function tsdiag() returns an object with the correct type of class, and that tsdiag returns an error when used on data that has more than one variable. I've also included a quick smoke test for xyplot here to demonstrate that xyplot doesn't work on arima models - this can be moved to a new PR (perhaps the upgraded test suite).